### ISSUE TITLE
[feature request] Customized Thumbnails

### ISSUE THREAD

**Issue by Feans** [k-ivan/visual-bookmarks-chrome#93]
Давно уже хотелось бы иметь такую возможность как-то влиять на то, что отображается в миниатюре. Часто в миниатюру попадает не то, что характеризует страницу, например вечнозагружающийся черный квадрат с видео, или огромная шапка сайта, основное содержимое которого находится под ней. Ну вот к примеру получается что добавив несколько ссылок на какой-нибудь видеосайт, все эти ссылки выглядят неразличимо - в них попала только огромная шапка сайта, а картинки с обложками фильмов не вошли или где-то в самом низу кусочек.

Я и наверное многие другие используют визуальные закладки как некое пространство в котором можно визуально быстро найти то, что нужно. Поэтому неплохо было бы дать возможно пользователю при создании миниатюры выбирать кусок сайта, который будет использован в качестве миниатюры. Такая функция встроена браузер Opera: при нажатии на добавлении закладки, появляется в верхнем правом углу окошко в котором показан уменьшенный скриншот сайта, который можно прокрутить по всем направлениям, уменьшить или увеличить, таким образом можно выбрать любой нужный кусок и после этого нажать кнопку "добавить", он и будет использовать в качестве миниатюры, либо если эта страница уже добавлена была в закладки то можно обновить миниатюру, выбрав новое изображение.

Кроме того в Опере в этой же функции, если нажать на стрелку рядом со скриншотом в этом всплывшем окошке, можно прокрутить различные изображения, которые есть на этой странице взятые из кода html и любое из них так же можно приблизить, удалить, прокрутить в стороны чтобы выбрать нужный кусок картинки для миниатюры.

В старом firefox было расширение Viewmarks, оно работало точно так же, можно было выбрать любой кусочек. Делался длинный скриншот страницы, в несколько экранов и его можно было до определенной степени прокрутить до нужного места вниз и увеличив это место сделать его миниатюрой.

В большинстве случаев конечно того функционала который сейчас есть - достаточно, однако я использую ваше расширение и как панель быстрого доступа к нужным сайтам (вместо Fast Dial), так и как коллекцию ссылок на фильмы, книги, музыку... поэтому важно и графическое исполнение, как скажем в папке музыка у меня будут обложки альбомов и я быстро найду то, что мне нужно. Ну как пример.  Я поэтому параллельно пользуюсь другими браузерами в которых есть такие расширения или встроенные функции. Но если будет в вашем расширении такая опция, которую я давно ищу для хрома, и которой пожалуй нет ни в одном расширении, это было бы круто.

Ну и еще один момент - можно сделать так, чтобы при добавлении новой закладки через меню расширения, миниатюра создавалась автоматически и даже можно просто скриншот открытого окна делать без всяких всплывающих окон.

**Comment by k-ivan** [comment]
Функционал довольно сложный, судя по описанию, может быть очень много подводных камней(точнее точно будет много). Мне кажется, предложения интересные, но сразу предупрежу, не ждите в скором времени.

**Comment by Feans** [comment]
![image](https://i.imgur.com/Sa0wTPs.gif)

Вот пример как это реализовано в браузере опера. Т.е. можно стрелочками прокрутить картинки со страницы или выбрать скриншот. И скриншот и картинки можно масштабировать колесиком мыши и передвигать внутри окна зажатой левой кнопкой мыши.

**Comment by k-ivan** [comment]
Согласен, выглядит круто! Мне нравится. Я не уверен насчет картинок, но со скриншотом точно можно попробовать. Взял на заметку. Спасибо за идею!

**Comment by Feans** [comment]
![image](https://c2n.me/4fWlS7R.jpg)

Так выглядело расширение Viewmarks для Firefox.
Кнопки под миниатюрой: обновить миниатюру,  удалить миниатюру, +/- масштабирование, знак дискеты - загрузить собственное изображение которое будет миниатюрой, и оставшиеся две кнопки скрывают расширенные опции справа: панель ключевых слов/тегов/информации и панель быстрого добавления в недавно использованные папки закладок (если по ним щелкнуть то сразу закладка добавится в нужную папку).

![image](https://c2n.me/4fWmkot.jpg)

Так выглядит окошко с выключенными расширенными панелями. У расширения было море настроек, например импорт/экспорт миниатюр в файл, фоновое создание пропущенных миниатюр, мгновенное добавление закладки с миниатюрой, создание миниатюр для всех закладок, настройка поведения при нажатии кнопки расширения, и много всего, что не перечислить. К сожалению после того как firefox выкатил новую версию в которой перестало работать большинство расширений, поддержка и многих других старых расширений прекратилась. Многолетний труд разработчиков расширений пропал впустую (не в первый раз кстати они заставляли переделывать расширения или вовсе писать с нуля из-за "новых технологий", такой мартышкин труд не всем придется по душе). С того момента база пользователей firefox значительно упала, и Chrome окончательно победил. Мало того, что старые расширения требовали обновления, так еще и был урезаны возможности которые было разрешено использовать в новой версии браузера, "идущего в ногу со временем". Что сделало просто невозможным создавать действительно многофункциональные расширения. Быстрый браузер, без расширений, требующий кучу памяти.

Боюсь Chrome движется по подобному пути.  Ни встроенных лент RSS, ни нормальных удобных панелей закладок, ни настроек proxy, ни возможности менять интерфейс и внешний вид, да много чего необходимого просто сознательно не существует.

**Comment by k-ivan** [comment]
> Боюсь Chrome движется по подобному пути.

Монополия никому на пользу не идёт. Ещё не забыт пример IE.

> Ни встроенных лент RSS, ни нормальных удобных панелей закладок, ни настроек proxy, ни возможности менять интерфейс и внешний вид, да много чего необходимого просто сознательно не существует.

Ну прокси есть, просто он идёт от настроек OS. А внешний вид скорее отдан на откуп сообществу. Но в целом сложно спорить, один только манифест3 чего стоит. Он сразу кучу всего перечеркнул, или как минимум заставил полностью переписать. Отобрали много, а заменили каким-то костылем.

**Comment by tipodanet** [comment]
Есть возможность из разряда "для продвинутых". Это уже реализовано. Там не расширение, там самодельный виджет, в Chrome Store его нет.
Там указывается id, или class,  и по-моему, размеры окна для preview.
Юзер должен открыть DevTools и найти этот  id, или class,  
Самый простой вариант. Т.к. прокрутки , наверное,  в Опере делала команда из многих людей. Мы не дождёмся от одного Ивана.
А с классом все понятно и легко

**Comment by tipodanet** [comment]
прочитал отзыв одного осторожного  слишком
" "Доступ к копируемым и вставляемым данным", очень настораживает. Пока прекращаю пользоваться, "
Иван , вы бы объяснили людям (выделить **RED BOLD** + underline ) , что расширение работает на своей странице, на Express Panel, а не на всех открытых. 
На всех открытых доступа к буферу не имеет . Или имеет  ? :) 
Хотя мне лично по барабану, имеет или нет. Мне главное, чтобы меньше  проблем с лишними кликами.

2. Если вы ходите нормальный feedback от юзеров, разместите этот GitHub в описании . Вряд ли кто-то будет искать его по отзывам или так напрягаться, чтобы писать на мыло. Даже я свой отзыв вижу без Вашего ответа. Ищу в общем порядке.

**Comment by k-ivan** [comment]
Здравствуйте!

Таких отзывов было не мало, кто-то оставлял отзыв в магазине расширений, кто-то написал на почту, с просьбой убрать.

Данное разрешение глобальное. Тут ведь ещё какой момент, почему люди обратили на это внимание, потому что браузер после обновления, "заботливо" отключил расширение и вывел страшное уведомление для пользователя, что отключил по причине того, что автор расширения запросил вот такое то право. Это к вопросу о жёсткости браузеров к разрешениям.
Сделал его опциональным по умолчанию, извините, включать по умолчанию не буду, кто захочет, сам активирует, кого беспокоит, просто откажутся. Изначально сделал по клику на этот пункт, без посещения настроек, но в итоге так оптимально и для расширения и для тех, кто не доверяет расширению.

По поводу ссылок на обратную связь, все хочу встроить ссылки в само расширение и все забываю. Нужно бы и текста в сторах обновить.

**Comment by tipodanet** [comment]
Конечно, если браузер начинает вредничать и отключать разширение, то все правильно. Я включаю все Галки, включая "Инкогнито". Потому что я им не пользуюсь :)

Иван, надо оживить главную страницу в Store. А первая должна быть видео, однозначно. Надо собрать красивых  закладок, выбрать стилей анимации повеселее. 💥. 
Чтобы красиво снять , надо какие-то редакторы специальные, заточенные под это. В Фильморе это сложно.

**Comment by tipodanet** [comment]
Все так дрожат за свои пароли. 🔑 сейчас, наверное, невозможно пользоваться паролями без программы типа Bitwarden,  у меня около около 1000 паролей. А сейчас с развитием генераторов паролей, если кто ими пользуется, так вообще невозможно без программы этих парольных ассистентов,  в каждом браузере обязательно есть менеджер паролей. Не знаю про это, можете меня проиинструктировать; когда менеджер паролеф подставляет эти 2 строки, он же не пользуется буфером обмена ? Тогда к чему эти тряски вокруг безопасности ?. Там, где касается карт,  100% идентификация через телефон . 
Распространённая паранойя, еще одна: где-то в ВК: Чтобы войти в поле ввода пароля , нужно ввести 1 пароль. А потом, после этого ввести номер телефона 📴, и  ввести пароль ещё раз.🔂. То есть телефон привязан к ВКонтакту, а логин привязан к Яндексу . И получается такая двухступенчатая Идентификация, а внутри ещё двухфакторная. Тут шансов добраться к паролю через буфер обмена, без доступа к телефону ( я считал ) примерно 0, 0000000001%.

**Comment by k-ivan** [comment]
> Иван, надо оживить главную страницу в Store. А первая должна быть видео, однозначно. Надо собрать красивых  закладок, выбрать стилей анимации повеселее. 💥.

Обязательно обновим.

**Comment by k-ivan** [comment]
> Все так дрожат за свои пароли. 🔑 сейчас, наверное, невозможно пользоваться паролями без программы типа Bitwarden,  у меня около около 1000 паролей. А сейчас с развитием генераторов паролей, если кто ими пользуется, так вообще невозможно без программы этих парольных ассистентов,  в каждом браузере обязательно есть менеджер паролей. Не знаю про это, можете меня проиинструктировать; когда менеджер паролеф подставляет эти 2 строки, он же не пользуется буфером обмена ? Тогда к чему эти тряски вокруг безопасности ?. Там, где касается карт,  100% идентификация через телефон . 
> Распространённая паранойя, еще одна: где-то в ВК: Чтобы войти в поле ввода пароля , нужно ввести 1 пароль. А потом, после этого ввести номер телефона 📴, и  ввести пароль ещё раз.🔂. То есть телефон привязан к ВКонтакту, а логин привязан к Яндексу . И получается такая двухступенчатая Идентификация, а внутри ещё двухфакторная. Тут шансов добраться к паролю через буфер обмена, без доступа к телефону ( я считал ) примерно 0, 0000000001%.

Небольшая просьба, давайте темы не относящиеся к данному ишью обсуждать, если требуется, в дискуссиях.
